### ISSUE TITLE
Cannot connect from some regions (Chicago/North EU/?)

### ISSUE THREAD

**Issue by jeanmichelmorin** [FriendsOfGalaxy/galaxy-integration-steam#127]
**Describe the bug**
GOG Steam randomly logged me out yesterday. I try to reconnect and it just spins for a long time then stops and ask me to retry again. I asked friends and checked Reddit, and they all have this issue with GOG Steam right now. It also doesn't ask me for my Steam Auth Code. Simply the username and the password (not even the Steam page).

**Attached logs**
(https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6754501/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

"Not connected, try again. ---- Disconnect"
![image](https://user-images.githubusercontent.com/44597109/124273661-4b0b7800-db0e-11eb-9ebe-fe99cda3dd70.png)

**Comment by adriaan147** [comment]
happened to me too

**Comment by UsefullPig** [comment]
Same here. I even tried to uninstall and reinstall the Steam plugin.

**Comment by CWCprime** [comment]
Same issue happening with me. I checked to see if some games I bought from Steam were added. They didn't appear, so I tried to sync integrations, and then *poof* all my games disappeared. I've tried disconnecting then reconnecting, reinstalling GOG Galaxy, nothing works. All other integrations that I use work fine.

**Comment by MadGenius792** [comment]
Same issue here !

**Comment by nodeunit** [comment]
Have disconnected and reconnected; GOG cannot query my Steam games anymore, and only my Steam games

**Comment by David341** [comment]
Same problem here

**Comment by Icemasta** [comment]
Same problem here. I've sometimes managed to actually log in, after several attempts. I added loggers throughout a local copy.

I dunno if it's issues on the steam side, it's very inconsistent if I get logged in or not. Once I am in, according to the logs, it successfully pulls my friend list, it also very quickly says it processed the imported game list. I added loggers to the function get_owned_games in particular. It does get run, it passes the authentication check, it does wait for the game cache to be ready, but "self._games_cache.get_owned_games()" is always empty when returned, so no games is ever actually added. 

Now, I do not know how wait_ready works, if wait_ready(90) means it waits 90 seconds and then times out and proceed, that might be the issue. I did not time it but there was a significant delay between the waiting and then resuming in my logs.

**Comment by Icemasta** [comment]
Alright, I've  tested quite a bit, it seems to be entirely on steam's side. If I am not mistaken after having gone through the code this morning, the issues mostly stem from Steam not responding in time. If it does respond to the login request, when it tries to pull the game, it times out due to wait_ready. I am currently experimenting with increasing the time outs and see if steam sends it along the way.

**Comment by jeanmichelmorin** [comment]
> Alright, I've tested quite a bit, it seems to be entirely on steam's side. If I am not mistaken after having gone through the code this morning, the issues mostly stem from Steam not responding in time. If it does respond to the login request, when it tries to pull the game, it times out due to wait_ready. I am currently experimenting with increasing the time outs and see if steam sends it along the way.

That was my initial thinking too. This git repository was last updated about 17 days ago. I also did some digging on my end and I'm pretty sure you're right. Hopefully it is fixable somehow.

**Comment by FriendsOfGalaxy** [comment]
Hi, thanks for all your input. I'm investigating.

To problems:
- (OP case) https://steamerrors.com/48 Maybe Steam had temporal problems with some CMs (auth servers), so it ask to use different one, but thep lugin does not handle that.
- https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/118 - what @Icemasta described - probably the plugin regression issue. You can try version 0.57 to check if the problem exists.

90 Timeout on owned_games is OK. The problem is the whole method fall into >120s which causes timeout on Galaxy side and no owned games are imported at all.

**Comment by ArctanAlpha** [comment]
Hi,

Not a programmer / coder at all. Created this account to say this:
1. Thanks for the hard work

2. Out of shear desperation I installed V0.21 (the oldest available on here) and it worked. Thought I will let you know for what it is worth.

Cheers,
ArctanAlpha

**Comment by CWCprime** [comment]
I also tried 0.21 and it works for me too!

**Comment by ChakraaArcana** [comment]
Hi!
Is there any guides on how to manually install older versions of the plugin so that I have access to my library for now?
Thanks!

**Comment by CWCprime** [comment]
@ChakraaArcana 
Try copying the contents of the 0.21 download folder into the Steam folder in C: > Users > Your name > AppData > Local > GOG.com > Galaxy > Plugins > Installed

**Comment by DrSomeone** [comment]
> @ChakraaArcana
> Try copying the contents of the 0.21 download folder into C: > Users > Your name > AppData > Local > GOG.com > Galaxy > Plugins > Installed > Steam folder
> 
> Then reconnect Steam in GOG Galaxy settings. That's how I got it to work for me.

How did you do this? I'm new to github, and don't know where to get this v0.21 and manually install it.

**Comment by CWCprime** [comment]
@DrSomeone 
Here's how I did it exactly. First, exit GOG Galaxy. Then, on your PC, go to:

C: > Users > Your name > AppData > Local > GOG.com > Galaxy > Plugins > Installed > Steam folder (will have a bunch of number & letters in the name)

And delete everything in that folder. This probably isn't necessary, but I just like to make sure there's nothing in there that might mess up the v0.21 install.

Then download v0.21 here: [https://github.com/FriendsOfGalaxy/galaxy-integration-steam/releases/tag/v0.21](url) Assuming you are Windows, download the "window.zip" archive.

Extract the folder and copy the contents into the same folder from above that we deleted the old plugin from. Then start GOG Galaxy again, and reconnect Steam. This is how I got it to work for me.

**Comment by Icemasta** [comment]
Wouldn't v0.21 immediately update to v0.59 unless you disable the auto-update feature though? Have you checked that the v0.21 that you've installed is still v0.21?

Because I tried that a few days ago, v0.57 as @FriendsOfGalaxy suggested still runs into the same issue, but I had to disable the upload process. When I tried v0.21, it would instantly update to v0.59, and I couldn't find how to disable the update feature in v0.21 because it wasn't in the manifest file, so it's using another method, which I couldn't isolate.

**Comment by ArctanAlpha** [comment]
Yes, the auto update function does indeed update the version and then the same problem exists for me.

After installing V0.21 I disabled auto update by changing the version to "10" in the "manifest.json" file before starting Galaxy again.

Cheers,

**Comment by bigboimarkus** [comment]
been having mutiple problems here aswell and it was just working fine before the summer sale went live

**Comment by Icemasta** [comment]
Hmm, I've tried the following versions:

v0.21: Failed to start.
v0.29: Plugin has crashed
v0.39: **Confirmed Partial working!** This one opens up the actual steam website to do the authentication. It imports the game library. **It fails to identify installed games.**
v0.42:  Same as v0.46.5
v0.46.5: Can login, similar to v0.39, fails to import game. 
v0.49: Same as v0.59
v0.57: Same as v0.59
v0.59: Can sometimes login if you fidget a bit with it (Connect -> Login -> 2FA -> ... -> Do not click retry, click disconnect -> Plugin fails -> Connect -> Do not agree the terms -> Log in -> 2FA -> Agree to terms -> Logged in -> Friend list is loaded, rest fails).

So the only thing I can say is that the new authentication process does not work with whatever Steam did, but the old method works flawlessly. v0.39 is consistent, it manages to import library, it just fails to properly identify installed games, or the way it feeds the data to GoG is outdated.

**Comment by FaithTheSlayer** [comment]
Started having this issue on july 3rd, fixed on 4th, fine on 5th, and today the 6th its really bad. Please fix!

**Comment by FriendsOfGalaxy** [comment]
Hi guys,

I've tried to find repro and looks like some Steam servers are affected, but not all the time.
For now I know about: `sto` and `ord` (Stokholm and Chicago steam servers). All Steam CM list: 
https://steamstat.us/

I mean those servers works OK for login to Steam app, but not for login to a plugin as the plugin does not understand eresult 48 `TryWithDifferentCM` error message.

For not I have no repro so I would like to ask you to test this version (windows only):
[win - try different cm.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6775745/win.-.try.different.cm.zip)

and if it still does not work then this one:
[win - try with different region cm.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6775750/win.-.try.with.different.region.cm.zip)

All steps for testing:
- shutdown Galaxy completely
- install a build according to the readme: https://github.com/FriendsOfGalaxy/galaxy-integration-steam#for-custom-build
- restart Galaxy
- wait to see how plugin is connecting (if was connected previously) or connect it from the beginning.

Please share logs if it works or not - it will be helpful to check how the logic behaves. Thanks!

**Comment by Icemasta** [comment]
Thanks, this patch works perfectly! Connected, loaded library, and loaded installed games.

**Comment by FriendsOfGalaxy** [comment]
@Icemasta thanks for letting me know. If you could also upload your logs it would be great. I would like to check if it helped for OP problem or it was a matter of coincidance.

Instruction where to find logs: 
https://github.com/FriendsOfGalaxy/galaxy-integration-steam/wiki/Log-files

**Comment by jeanmichelmorin** [comment]
> Hi guys,
> 
> I've tried to find repro and looks like some Steam servers are affected, but not all the time.
> For now I know about: `sto` and `ord` (Stokholm and Chicago steam servers). All Steam CM list:
> https://steamstat.us/
> 
> I mean those servers works OK for login to Steam app, but not for login to a plugin as the plugin does not understand eresult 48 `TryWithDifferentCM` error message.
> 
> Currently I have no reproduction so I would like to ask you to test this version (windows only):
> [win - try different cm.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6775745/win.-.try.different.cm.zip)
> 
> and if it still does not work then this one:
> [win - try with different region cm.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6775750/win.-.try.with.different.region.cm.zip)
> 
> All steps for testing:
> 
> * shutdown Galaxy completely
> * install a zipped build according to the readme: https://github.com/FriendsOfGalaxy/galaxy-integration-steam#for-custom-build
> * restart Galaxy
> * wait to see how plugin is connecting (if was connected previously) or connect it from the beginning.
> 
> Please share logs regardless of if it works for you or not - it will be helpful to check how the logic behaves.
> https://github.com/FriendsOfGalaxy/galaxy-integration-steam/wiki/Log-files
> 
> Thanks!

The first one worked fine. It generated 2 log files, one seems to be about imported games, but I will share it with you regardless just incase it may be useful.

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.txt](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6777022/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.txt)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8 (2).txt](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6777024/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.2.txt)

Here's some quicksteps for people struggling:

- Close Galaxy.

- Go to **%localappdata%\GOG.com\Galaxy\plugins\installed\** and delete the **steam_ca27391f-2675-49b1-92c0-896d43afa4f8**
folder.

- In **%localappdata%\GOG.com\Galaxy\plugins\installed\** , create a folder named **steam** (name doesn't matter). 

- Download this zip file: [win - try different cm.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6775745/win.-.try.different.cm.zip)

- Unzip the content of the downloaded zip file into the folder we just created **%localappdata%\GOG.com\Galaxy\plugins\installed\steam**

- Find the **manifest.json** file and change the version variable to 9.9. **"version": "9.9",**

And then you should be good to go! Launch GOG Galaxy 2.0 and try connecting it to Steam.

**Comment by FriendsOfGalaxy** [comment]
@jeanmichelmorin thanks for logs.

Looks like the problem fixed itself on Steam side. If you can, please share logs in a few hours again just restarting galaxy a few times. Each time may randomize received servers list so we can see it if happens.

FYI: you're affected by Chicago `ord` server, but sometimes also connect to Sterling `iad` one which works ok.

I'm waiting for more tests before proceeding with a release.

**Comment by FaithTheSlayer** [comment]
got mine to work yesterday i think it was 0.27 or 28 and changed the version number in json to 10.0 so it wouldn't auto update

**Comment by jeanmichelmorin** [comment]
@FriendsOfGalaxy I just tried to login again, and still works. Here are my new logs.

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8-1.log](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6780216/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8-1.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8-2.log](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6780217/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8-2.log)

**Comment by ninja-gaijin** [comment]
I had an issue of not syncing, not adding new games, tracking playtime, achievements etc, only on steam. 
The first linked version worked for me to update my games (it added newly purchased) but it didn't add playtime or achievements.
[GalaxyClient.log.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6786204/GalaxyClient.log.zip)

**Comment by FriendsOfGalaxy** [comment]
@ninja-gaijin that is a different problem, I'll create separate issue for that.

Current issue status: Steam fixed the problem on their side. Nevertheless the potential fix for this kind of problems will be added to plugin soon.

**Comment by OttoWerse** [comment]
I am in Central EU and I still get this issue. I got logged out today with no way to sign back in. It loads forever, and then just get's stuck on "offline. retry" but clicking retry doesn't seem to do anything. I do sometimes get SteamGuard promts, sometimes not. I think I have tired every possible troubleshooting step besides reinstalling Steam or Windows. Reinstalling Galaxy itself (including deleting all files in ProgramData and AppData) has helped. I am thinking maybe this is related to the recent big Steam update?

**Comment by jeanmichelmorin** [comment]
> I am in Central EU and I still get this issue.
> I got logged out today with no way to sign back in. It loads forever, and then just get's stuck on "offline. retry" but clicking retry doesn't seem to do anything. I do sometimes get SteamGuard promts, sometimes not.
> 
> I think I have tired every possible troubleshooting step besides reinstalling Steam or Windows. Reinstalling Galaxy itself (including deleting all files in ProgramData and AppData) has helped. I also tried disabling Ipv6 for my Ethernet controller, but sadly that didn't change anything either.
> 
> I am thinking maybe this is related to the recent big Steam update?

Hello, try following these steps to see if it fixes your issue. 
https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/127#issuecomment-875557839

This is only a temp fix and not a patch, as your GOG Steam will be out of date.

**Comment by FriendsOfGalaxy** [comment]
> I am in Central EU and I still get this issue.
> I got logged out today with no way to sign back in. It loads forever, and then just get's stuck on "offline. retry" but clicking retry doesn't seem to do anything. I do sometimes get SteamGuard promts, sometimes not.
> 
> I think I have tired every possible troubleshooting step besides reinstalling Steam or Windows. Reinstalling Galaxy itself (including deleting all files in ProgramData and AppData) has helped. I also tried disabling Ipv6 for my Ethernet controller, but sadly that didn't change anything either.
> 
> I am thinking maybe this is related to the recent big Steam update?

Hey @OttoWerse that's interesting that this is still occurring for you. Could you please provide your logs? See https://github.com/FriendsOfGalaxy/galaxy-integration-steam/wiki/Log-files

**Comment by FriendsOfGalaxy** [comment]
I'm closing as the fix was released and there is no more active signs of the problem. 

However it is worth to note, that the problem with handling eresult 48 is NOT fixed when logging for the first time. Fixing that requires profound refactor for overcomplicated authentication flow and error handling along the way. Planned, but not a priority right now.
